llvm/lib/Target/WebAssembly/WebAssemblyAsmPrinter.h

```diff
@@ -28,6 +33,9 @@ class LLVM_LIBRARY_VISIBILITY WebAssemblyAsmPrinter final : public AsmPrinter {
   WebAssemblyFunctionInfo *MFI;
   bool signaturesEmitted = false;
 
+  // vec idx == local func_idx
+  std::vector<std::optional<BranchHintRecord>> branchHints;
```

Comment:
Use SmallVector. Why wrap it in optional, shouldn't an empty hint vector indicate the same?